perk-store-webapps/src/main/webapp/vue-app/components/perk-store/BuyModal.vue

```diff
@@ -127,6 +136,25 @@ export default {
     isMetamaskWallet() {
       return window.walletSettings.wallet?.provider === 'METAMASK';
     },
+    isProductOwner() {
+      return this.product.userData.canEdit || this.product.userData.username === this.product.creator.id || this.product.userData.username === this.product.receiverMarchand.id ;
+    },
```

Comment:
the user can buy his own product that he creates if the Receiver Wallet Merchant is different from his own wallet. Thus you just need @ArfaouiMohamedAmine to check on this condition:
* **Can buy product** = Product has a Merchant Wallet AND the Merchant Wallet is different from current user's wallet.

```javascript
    canBuyProduct() {
      return this.product?.receiverMarchand?.id && this.product?.userData?.username !== this.product.receiverMarchand.id;
    },
```